[BOP-775] RBAC migration (#86)
Mirantis/mke-docs: content/docs/migrate-from-MKE-3.md

In the event of an upgrade failure, the upgrade process rolls back,
restoring the MKE 3 cluster to its original state.

## RBAC Migrations

> MKE 4 uses a different RBAC model than MKE 3. MKE 4 no longer supports swarm mode
> and uses Kubernetes RBAC for authorization. As a result, the RBAC configuration
> for Swarm mode does not exist in MKE 4. However, the Swarm mode roles have always
> had Kubernetes RBAC roles associated with them. The upgrade process migrates these
> Kubernetes roles and bindings to the new MKE 4 cluster.

Reviewer: ```suggestion
The RBAC model MKE 4 uses differs significantly from the one in use by MKE 3.

As MKE 4 no longer supports Swarm mode, the platform uses Kubernetes RBAC for authorization. Thus, the RBAC configuration for Swarm mode does not exist in MKE 4. The Swarm mode roles have Kubernetes RBAC roles associated with them, however, and the MKE 4 upgrade process migrates these
Kubernetes roles and bindings to the new MKE 4 cluster.
```

> The following sections list the MKE 3 Swarm mode roles and their corresponding

Reviewer: ```suggestion
The sections that follow detail the MKE 3 Swarm mode roles and their corresponding
```

Kubernetes RBAC permissions:

> ### View Only

Reviewer: I suggest that the #### heading level be a dropdown heading that expands when clicked to reveal the bullet detail.

I found information on how to do this in markdown here: https://gist.github.com/citrusui/07978f14b11adada364ff901e27c7f61

Thus, the ### headings remain, the #### headings become the text that contains the dropdown text, and the dropdown text will be the bullet lists.

Author: Done

- ConfigMap
- ControllerRevision
- CronJob
- CustomResourceDefinition
- DaemonSet
- Deployment
- Endpoint
- Event
- ExternalAdmissionHookConfiguration
- HorizontalPodAutoscaler
- Ingress
- InitializerConfiguration
- Job
- LimitRange
- NetworkPolicy
- PersistentVolumeClaim
- PersistentVolume
- PodDisruptionBudget
- PodPreset
- Pod
- PodTemplate
- ReplicaSet
- ReplicationController
- RoleBinding
- Role
- Secret
- ServiceAccount
- Service
- Stack
- StatefulSet
- ThirdPartyResource

### Full Control

#### get, list, watch, create, delete, patch, update

- CertificateSigningRequest
- ClusterRoleBinding
- ClusterRole
- ComponentStatus
- ConfigMap
- ControllerRevision
- CronJob
- CustomResourceDefinition
- DaemonSet
- Deployment
- Endpoint
- Event
- ExternalAdmissionHookConfiguration
- HorizontalPodAutoscaler
- Ingress
- InitializerConfiguration
- Job
- LimitRange
- Namespace
- NetworkPolicy
- Node
- PersistentVolumeClaim
- PersistentVolume
- PodDisruptionBudget
- PodPreset
- Pod
- PodTemplate
- ReplicaSet
- ReplicationController
- ResourceQuota
- RoleBinding
- Role
- ServiceAccount
- Service
- Stack
- StatefulSet
- StorageClass
- ThirdPartyResource
- User

### Scheduler

> No kubernetes RBAC permissions are associated with the Scheduler role.

Reviewer: ```suggestion
No Kubernetes RBAC permissions correlate to the Scheduler role.
```
